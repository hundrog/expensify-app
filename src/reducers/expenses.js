// Expenses Reducer
const expensesReducerDefaultState = [];

const expensesReducer = (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case "ADD_EXPENSE":
      return [...state, action.expense]; //array spreading, take an array and add at the end or begining
    case "REMOVE_EXPENSE":
      // So... we take the expense, destructure the id, then compare the id to filter it
      return state.filter(({ id }) => id !== action.id);
    case "EDIT_EXPENSE":
      return state.map(expense => {
        if (expense.id === action.id) {
          // object spreading, take the object the override
          return {
            ...expense,
            ...action.updates
          };
        } else {
          return expense;
        }
      });
    default:
      return state;
  }
};

export default expensesReducer;
