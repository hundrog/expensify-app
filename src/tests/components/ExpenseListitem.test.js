import React from "react";
import { shallow } from "enzyme";
import ExpenseListItem from "../../components/ExpenseListItem";
import expenses from "../fixtures/expenses";

test("should render expense list item correctly", () => {
  const expense = expenses[1];
  const wrapper = shallow(<ExpenseListItem id={expense.id} {...expense} />);
  expect(wrapper).toMatchSnapshot();
});
