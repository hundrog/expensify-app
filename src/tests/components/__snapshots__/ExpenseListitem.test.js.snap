// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item correctly 1`] = `
<div>
  <Link
    to="/edit/2"
  >
    <h3>
      soda
    </h3>
  </Link>
  <p>
    amount: 
    800
     | created at: 
    -345600000
  </p>
</div>
`;
